### PR TITLE
Batch Normalization support for Net2Net

### PR DESCRIPTION
**Pull request by dasguptar**
Added batchnorm cases for both wider and deeper functions.
Added corresponding tests, and cleaned up existing tests.
Removed batchnorm warning.
Updated README.

### CODE REVIEW

File: test.lua [hunk 1]
```diff
@@ -17,57 +17,107 @@ local inp = torch.randn(4, 100)
...
 
 local outWider = m:forward(inp):clone()
 
 assert(out:add(-1, outWider):abs():max() < eps)
 
+-- insert BatchNormalization layer after linear layer
+local m = nn.Sequential()
+m:add(nn.Linear(100,200))
+m:add(nn.BatchNormalization(200))
+m:add(nn.ReLU())
```
Comment (soumith) [inline test.lua:31]: if there's a batchnorm here as well, just error out.

**Comment by soumith** [comment]
fix tabs / indentation, fix typos. otherwise looks good.

**Comment by soumith** [comment]
let's merge this in soon...

**Comment by dasguptar** [comment]
Fixed indentation and typos. Squashed commits. Have a look if this is good to merge.

**Comment by soumith** [comment]
Thanks!
